debezium-core/src/main/java/io/debezium/heartbeat/HeartbeatImpl.java

```diff
@@ -68,12 +68,18 @@
     @Override
     public void heartbeat(Map<String, ?> partition, Map<String, ?> offset, BlockingConsumer<SourceRecord> consumer) throws InterruptedException {
         if (heartbeatTimeout.expired()) {
-            LOGGER.debug("Generating heartbeat event");
-            consumer.accept(heartbeatRecord(partition, offset));
+            forcedBeat(partition, offset, consumer);
             heartbeatTimeout = resetHeartbeat();
         }
     }
 
+    @Override
+    public void forcedBeat(Map<String, ?> partition, Map<String, ?> offset, BlockingConsumer<SourceRecord> consumer)
+            throws InterruptedException {
+        LOGGER.debug("Generating heartbeat event");
+        consumer.accept(heartbeatRecord(partition, offset));
```

Comment:
Thinking about it, it seems we'd emit a message here also if heartbeating is disabled, right? I don't think we should do this, instead this should only be emitted if heartbeating is enabled (i.e. the "force" aspect of this would just be to ignore the timeout expiration).

Reply:
IMHO this is covered with https://github.com/debezium/debezium/pull/606/files#diff-b911a5a9bf14cb8afe18a66e5f26c1f7R59

Reply:
@jpechane That linked line is an empty line, I'm not sure what you are saying?

Reply:
@gunnarmorling, ok, please check the method starting on the line below

Reply:
Ah, gotcha, that's the no-op impl. Yes, it looks legit then. Thanks.